src/lib/util/debug.js

```diff
@@ -24,27 +24,60 @@
  *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  **/
 
+/**
+ * Utility module to handle debugging.
+ * @module util/debug
+ * @see module:util/debug
+ */
+
 'use strict';
 
 const
 	_ = require('lodash'),
 	coreDebug = require('debug'),
-	debugStream = require('debug-stream'),
-
-	addBufferFormatter = (debug) => {
+	debugStream = require('debug-stream');
 
-		coreDebug.formatters.b = (buffer) => {
-			const lines = _.compact(_.split(buffer, '\n'));
-			_.each(_.initial(lines), (value) => {
-				debug(value);
-			});
-			return _.last(lines) || '';
-		};
+/**
+ * Adds the buffer formatter to the debugger.
+ * @instance
+ * @param {*} debug - The debug instance
+ */
+function addBufferFormatter(debug) {
 
+	coreDebug.formatters.b = (buffer) => {
+		const lines = _.compact(_.split(buffer, '\n'));
+		_.each(_.initial(lines), (value) => {
+			debug(value);
+		});
+		return _.last(lines) || '';
 	};
 
+}
+
+function enableLogging(config, namespace) {
+
+	const log = coreDebug(namespace);
+
+	if (!_.get(config, 'silent', false) && _.get(config, 'debug', false)) {
```

Comment:
`_.get` is good when you're trying to pull out a nested property value and you don't want to null check all of the steps in the path. But if it's a top-level property, I think it harms readability. I reckon you want
```javascript
if (config.debug && !config.silent) { //positive case first, because we're positive people with a positive outlook
```